lib/graphql_schema.rb

```diff
@@ -2,6 +2,24 @@
 require 'set'
 
 class GraphQLSchema
+  INVALID_SCHEMA = Class.new(StandardError)
+
+  class << self
+    def load_schema(schema)
+      case schema
+      when Hash
+        new(schema)
+      when GraphQLSchema
+        schema
+      when Pathname, String
+        schema = File.read(schema) if Pathname(schema).file?
```

Comment:
Correct about the behaviour. The only downside to `parse` and `read` is we kind of lose a place for dealing with an instance of `GraphQLSchema` itself. And maybe even a Hash? Since it's already "parsed".

What about a compromise:

```ruby
def load_schema(schema)
      case schema
      when Hash
        new(schema)
      when GraphQLSchema
        schema
      when String
        new(JSON.parse(schema))
      when Pathname
        new(JSON.parse(File.read(schema))
      else
        raise INVALID_SCHEMA
      end
    end
```

This just drops the string filename cause which I'm fine with.